### ISSUE TITLE
Build failure on multiple jumbo files

### ISSUE THREAD

**Issue by saltama** [ungoogled-software/ungoogled-chromium-macos#28]
Building Chromium on the latest Mojave/Xcode11 (the Xcode release shouldn't matter):

```
In file included from gen/chrome/browser/ui/ui_jumbo_mm.mm:52:
./../../chrome/browser/ui/cocoa/renderer_context_menu/render_view_context_menu_mac_cocoa.mm:101:42: error: too many arguments to function call, expected 0, have 5
  g_original_populatemenu_implementation(self, _cmd, menu, remainingEntries,
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.

In file included from gen/chrome/browser/browser_jumbo_mm.mm:5:
./../../chrome/browser/app_controller_mac.mm:317:60: error: too many arguments to function call, expected 0, have 5
  g_swizzle_imk_input_session->GetOriginalImplementation()(self, _cmd, range,
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~
1 error generated.
```

I didn't really look into these errors and i'm just modifying those function calls as required(still compiling).
I could submit a patch but... did someone else see these errors?

/cc @Zoraver @kramred

**Comment by kramred** [comment]
I did not see these errors. Did you checkout git tag `77.0.3865.90-1.1` in both `ungoogled-chromium-macos` and `ungoogled-chromium`?

I've used
```
# xcodebuild -version
Xcode 10.3
Build version 10G8
# sw_vers
ProductName: Mac OS X
ProductVersion: 10.14.6
BuildVersion: 18G95
```

Your errors could be related to Xcode 11, although as you noted
> Xcode release shouldn't matter

(Apple doesn't seem to have much love for non-mobile OS/software lately)

**Comment by Zoraver** [comment]
I did not see these errors either when building 77.0.3865.90.

**Comment by saltama** [comment]
> Did you checkout git tag 77.0.3865.90-1.1

No, i checked the latest commits and then i didn't and just went with master, trying again now, thanks.

**Comment by saltama** [comment]
Nope, those errors are still there, it's a linker issue it seems (fixing the calls doesn't help, still can't link the binaries) and at this point it's possible it could really be caused by the command line tools shipped with Xcode 11 (just released).

**Comment by dajmon** [comment]
I did not see these errors either.

`$ xcodebuild -version
Xcode 11.0
Build version 11A420a`

**Comment by saltama** [comment]
No idea of what's happening, i've tried a few permutation of tags/reverts to no avail.
I'm closing this since i'm the only one seeing it, i'll try again.

**Comment by hellotanuky** [comment]
Failed to compile on macOS 10.15 Beta 10(19A578c)/Xcode 11 GM(11A1027).

```
./../../chrome/browser/app_controller_mac.mm:317:60: error: too many arguments to function call, expected 0, have 5
  g_swizzle_imk_input_session->GetOriginalImplementation()(self, _cmd, range,
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~
1 error generated.
```

Last version (Chromium 76.x) was unable to compile too because of some error in headers provided by Xcode. Maybe this is a bug caused by Xcode?

**Comment by hellotanuky** [comment]
I tried Xcode 11.1, 11.2 beta 2 on macOS 10.15 and still got the same error.
```
./../../chrome/browser/ui/cocoa/renderer_context_menu/render_view_context_menu_mac_cocoa.mm:101:42: error: too many arguments to function call, expected 0, have 5
  g_original_populatemenu_implementation(self, _cmd, menu, remainingEntries,
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
It seems to be related to `IMP` prototype check in LLVM. I tried pass `-DOBJC_OLD_DISPATCH_PROTOTYPES` to clang++ and manually build the failed jumbo source file (e.g ui_jumbo_mm which contains `render_view_context_menu_mac_cocoa.mm`) and the build succeeded with no error. 

This compilation error never occured in any previous version and there's nothing to do with Xcode version since the build system uses its own LLVM binary in `third_party/llvm-build`.

**Comment by NikolaF94** [comment]
Ive got the same problem.
xcodebuild -version
Xcode 11.1
Build version 11A1027
sw_vers
ProductName:	Mac OS X
ProductVersion:	10.15
BuildVersion:	19A583

**Comment by kjcommon138** [comment]
I also got the same issue as well on:
```bash
>>> xcodebuild -version
Xcode 11.2
Build version 11B52
>>> sw_vers
ProductName:    Mac OS X
ProductVersion: 10.14.6
BuildVersion:   18G95
```

The way I worked around this error was to manually create a patch for `chrome/browser/app_controller_mac.mm` and `chrome/browser/ui/cocoa/renderer_context_menu/render_view_context_menu_mac_cocoa.mm` so that the extra arguments are removed from these two function calls.  Make sure to add the path to these patches to the `ungoogled-chromium/patches/series` file.  Beware that the path to these two .mm files also need to be included in `ungoogled-chromium/domain_substitution.list` before starting the build or else it'll fail due to the way that patches are applied using this whitelist to ensure that certain files can be patched and not others.

Unfortunately, the build failed anyway due to this issue: https://github.com/ungoogled-software/ungoogled-chromium-macos/issues/31#issue-506317150

```
[20797/20953] ACTION //tools/v8_context_snapshot:generate_v8_context_snapshot(//build/toolchain/mac:clang_x64)
FAILED: v8_context_snapshot.bin 
python ../../build/gn_run_binary.py ./v8_context_snapshot_generator --output_file=v8_context_snapshot.bin
./v8_context_snapshot_generator failed with exit code -5
ninja: build stopped: subcommand failed.
```
